controller/rq_hover.ml

```diff
@@ -12,10 +12,21 @@ let build_ind_type mip = Inductive.type_of_inductive mip
 
 type id_info =
   | Notation of Pp.t
-  | Def of (Pp.t * Names.Constant.t option * string option)
+  | Def of (Pp.t * Pp.t * Names.Constant.t option * string option)
```

Comment:
Might be a good idea to make this a record and give it some proper field names. It would make it easier to read and understand.

Reply:
All the code here was mostly placeholder as to test hover and wasn't properly designed, however it seems we got stuck with it for a while, due to other priorities.

If you look at the code, it is indeed pretty horrible haha. The Coq API doesn't absolutely help tho.

Reply:
Yeah, we should really try to upstream some better API changes on the Coq side. I believe it isn't considered a priority due to the mostly internal nature.